src/store/userSlice.ts

```diff
@@ -12,6 +12,7 @@ export interface UserSlice {
   account: Horizon.AccountResponse | undefined;
   isFunded: boolean | undefined;
   balances: Map<string, bigint>;
+  hasTrustline: Map<string,boolean>;
```

Comment:
is there a reason we are placing this in state?

there is a function called `requireTrustline` in `horizon.ts` that will return a boolean from the `account` and `Asset` on the Reserve - we could probably just use that directly in the borrow anvil?